vpr/src/noc/xy_routing.cpp

```diff
@@ -0,0 +1,170 @@
+
+#include <vector>
+
+#include "xy_routing.h"
+#include "globals.h"
+#include "vpr_error.h"
+
+XYRouting::~XYRouting() {}
+
+void XYRouting::route_flow(NocRouterId src_router_id, NocRouterId sink_router_id, std::vector<NocLinkId>& flow_route, const NocStorage& noc_model) {
+    // keep track of whether a legal route exists between the two routers
+    bool route_exists = true;
+
+    // get the source router
+    const NocRouter src_router = noc_model.get_single_noc_router(src_router_id);
+
+    // keep track of the last router in the route as we build it. Initially we are at the start router, so that will be the current router
+    NocRouterId curr_router_id = src_router_id;
+
+    // lets get the sink router
+    const NocRouter sink_router = noc_model.get_single_noc_router(sink_router_id);
+
+    // get the position of the sink router
+    int sink_router_x_position = sink_router.get_router_grid_position_x();
+    int sink_router_y_position = sink_router.get_router_grid_position_y();
+
+    /**
+     * Keeps track of which routers have been reached already
+     * while traversing the NoC. This variable will help determine
+     * cases where a route could not be found and the algorithm is
+     * stuck going back and forth between routers it has already
+     * visited.
+     * 
+     */
+    std::unordered_set<NocRouterId> visited_routers;
+
+    // clear any previously stored route before finding the current route
+    flow_route.clear();
+
+    /*
+     * If we are not at the sink router then run another iteration of the XY routing algorithm
+     * to decide which link to take and also what the next router will be in the path.
+     *
+     * */
+    while (curr_router_id != sink_router_id) {
+        // store the router that is currently visited at each iteration of the algorithm
+        const NocRouter curr_router = noc_model.get_single_noc_router(curr_router_id);
+
+        // get the position of the current router
+        int curr_router_x_position = curr_router.get_router_grid_position_x();
+        int curr_router_y_position = curr_router.get_router_grid_position_y();
+
+        // determine how we should path next
```

Comment:
"how we should path next" --> find direction to travel in next 
(or cut this comment, as pretty clear func name.

Reply:
comment removed